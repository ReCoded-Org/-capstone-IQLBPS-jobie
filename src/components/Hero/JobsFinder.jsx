import { useState } from "react";
import { filterData } from "../../data/filterData";

function JobsFinder({ setData }) {
  const [text, setText] = useState("");
  const handleChange = (e) => setText(e.target.value);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (text === "") {
      console.log("first");
    } else {
      setData(
        filterData.filter((item) => {
          if (item.position.toLowerCase().includes(text.toLocaleLowerCase())) {
            console.log(item);
            return item;
          }
          return null;
        })
      );
    }
  };
  return (
    <div className="App  bg-gray-200  flex-col justify-around  py-[50px]  ">
      <div className="flex  flex-col">
        <div>
          <h1 className="text-3xl font-medium text-dark-gray leading-loose ">
            {" "}
            Jobs Finder
          </h1>
          <p className="decoration-solid pt-0 text-dark-gray leading-loose">
            {" "}
            Find the best jobe for you here
          </p>
        </div>
        <div className=" flex-col mt-2">
          <form onSubmit={handleSubmit}>
            <input
              type="search"
              placeholder="Type to search "
              className="rounded-3xl px-6 py-2 outline-none relative left-8"
              value={text}
              onChange={handleChange}
            />
            <button
              type="submit"
              className="py-2 bg-accent px-5 rounded-3xl text-white relative   hover:bg-red-500 "
            >
              {" "}
              Search
            </button>
          </form>
        </div>
      </div>
    </div>
  );
}

export default JobsFinder;
