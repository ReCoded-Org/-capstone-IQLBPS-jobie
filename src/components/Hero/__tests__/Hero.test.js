import renderer from "react-test-renderer";
import Hero from "../Hero";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
  Trans: ({ children }) => children,
}));
it("renders correctly when the component matches the snapshot", () => {
  const tree = renderer.create(<Hero />).toJSON();
  expect(tree).toMatchSnapshot();
});
