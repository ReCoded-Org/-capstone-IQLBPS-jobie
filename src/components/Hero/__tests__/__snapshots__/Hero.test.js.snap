// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when the component matches the snapshot 1`] = `
<div
  className="bg-dark-gray flex justify-around items-center px-32 small:px-14 small:py-5 medium:py-6 medium:px-16 large:py-8 large:px-18"
>
  <div
    className="text-left"
  >
    <h1
      className="text-white text-5xl leading-snug font-semibold font-inter small:text-lg medium:text-3xl large:text-4xl"
    >
      10,254 Jobs 
      <br />
      Are Listed Here!
    </h1>
    <h3
      className="text-xl text-white medium:text-yellow-30 font-inter small:text-xs medium:text-lg large:xl"
    >
      find-your-job
    </h3>
     
    <div
      className=" mt-5 w-48 small:w-36 medium:w-56 large:w-72"
    >
      <input
        className="rounded-3xl px-6 py-2 outline-none absolute w-72 small:w-32 small:py-1.5 small:px-3 small:text-xs medium:w-52"
        placeholder="Type-to-search"
        type="search"
      />
      <button
        className="py-2 bg-accent px-5 rounded-3xl text-white relative  left-52 hover:bg-red-500 small:py-1.5 small:px-3 small:text-xs small:left-24 medium:left-36 "
        type="submit"
      >
        search
      </button>
    </div>
  </div>
  <div>
    <img
      alt="a girl searching for job"
      className="w-500 small:w-300 ml-7"
      src="/assets/jobsearch.png"
    />
  </div>
</div>
`;
